src/mame/machine/pgmprot_igs027a_type3.h

```diff
@@ -30,34 +18,46 @@ class pgm_arm_type3_state : public pgm_state
 	void init_killbldp();
 	void init_dmnfrnt();
 	void init_happy6();
-	DECLARE_MACHINE_START(pgm_arm_type3);
-	DECLARE_WRITE32_MEMBER( svg_arm7_ram_sel_w );
-	DECLARE_READ32_MEMBER( svg_arm7_shareram_r );
-	DECLARE_WRITE32_MEMBER( svg_arm7_shareram_w );
-	DECLARE_READ16_MEMBER( svg_m68k_ram_r );
-	DECLARE_WRITE16_MEMBER( svg_m68k_ram_w );
-	DECLARE_READ16_MEMBER( svg_68k_nmi_r );
-	DECLARE_WRITE16_MEMBER( svg_68k_nmi_w );
-	DECLARE_WRITE16_MEMBER( svg_latch_68k_w );
-	DECLARE_READ16_MEMBER( svg_latch_68k_r );
-	DECLARE_READ32_MEMBER( svg_latch_arm_r );
-	DECLARE_WRITE32_MEMBER( svg_latch_arm_w );
-	void svg_basic_init();
-	void pgm_create_dummy_internal_arm_region(int size);
-	void pgm_patch_external_arm_rom_jumptable_theglada(int base);
-	void pgm_create_dummy_internal_arm_region_theglad(int is_svg);
-	void pgm_descramble_happy6(uint8_t* src);
-	void pgm_descramble_happy6_2(uint8_t* src);
-	void svg_latch_init();
-	DECLARE_READ32_MEMBER( dmnfrnt_speedup_r );
-	DECLARE_READ16_MEMBER( dmnfrnt_main_speedup_r );
-	DECLARE_READ32_MEMBER( killbldp_speedup_r );
-	DECLARE_READ32_MEMBER( theglad_speedup_r );
-	DECLARE_READ32_MEMBER( happy6_speedup_r );
-	DECLARE_READ32_MEMBER( svg_speedup_r );
-	DECLARE_READ32_MEMBER( svgpcb_speedup_r );
-	DECLARE_MACHINE_RESET(pgm_arm_type3_reset);
+
 	void pgm_arm_type3(machine_config &config);
+private:
+	// svg
+	u8           m_ram_sel;
+	std::unique_ptr<u32[]>      m_shareram[2];    //for 5585G MACHINE
+
+	u32        m_latchdata_68k_w;
+	u32        m_latchdata_arm_w;
+	required_shared_ptr<u32> m_arm_ram;
+	required_shared_ptr<u32> m_arm_ram2;
+	required_region_ptr<u32> m_armrom;
+
+	optional_device<cpu_device> m_prot;
+
+	virtual void machine_start() override;
+	virtual void machine_reset() override;
```

Comment:
It's generally not a good idea to make overridden protected base members private.  You want to make it easy to make derived machines classes, and the derived classes should call the overridden methods from their immediate base.  If a derived class needs to call past the immediate base to a base of that, it's usually indicative of bad design somewhere.  Making overridden protected base methods private encourages this kind of stuff.